### PR TITLE
docsite: bump antsibull-docs, reference collection env var index

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
As discussed in the DaWGs meeting.

The collection env var index looks like this: https://ansible.fontein.de/collections/environment_variables.html. It explicitly does not contain any env variable mentioned in https://docs.ansible.com/ansible/devel/reference_appendices/config.html#environment-variables.

Other relevant antsibull-docs changes:
- Automatically use a module's or plugin's short description as the "See also" description if no description is provided (https://github.com/ansible-community/antsibull-docs/issues/64, https://github.com/ansible-community/antsibull-docs/pull/74).
- Remove support for ``forced_action_plugin``, a module attribute that was removed during the development phase of attributes (https://github.com/ansible-community/antsibull-docs/pull/63).
- Stop mentioning the version features were added for Ansible if the Ansible version is before 2.7 (https://github.com/ansible-community/antsibull-docs/pull/76).
- Callback plugins: show type, render indexes of callback plugins per type (https://github.com/ansible-community/antsibull-docs/pull/90)

Potential more new features that would require a new antsibull-docs release:
 - ~~https://github.com/ansible-community/antsibull-docs/pull/65 - requires a core team decision on https://github.com/ansible/ansible/pull/79370 (will be discussed by core team internally next week)~~ (see https://github.com/ansible/ansible/pull/79370#issuecomment-1404251346)
 - ~~https://github.com/ansible-community/antsibull-docs/pull/75 - requires a core team decision on https://github.com/ansible/ansible/pull/79471~~ won't happen anytime soon

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docsite

### CODE REVIEW

**Review by oraNod — APPROVED** [review]

**Comment by felixfontein** [comment]
Rebasing to re-trigger tests with latest version. This PR is now ready for review/merging from my side.

**Comment by oraNod** [comment]
@felixfontein Do you think this is ready or should we wait for more approvals?

**Comment by samccann** [comment]
It's worth staging the PR before we merge.

**Comment by felixfontein** [comment]
Yes, staging is definitely a good idea before merging.

**Comment by samccann** [comment]
@oraNod how do we find the link to the staged docsite from that??

**Comment by oraNod** [comment]
> @oraNod how do we find the link to the staged docsite from that??

@samccann I already took the runner offline but you can dm me. To be on the safe side I won't put the public hostname in a comment.

**Comment by samccann** [comment]
@felixfontein - should I be seeing this page when I stage this PR? - https://ansible.fontein.de/collections/environment_variables.html ?

**Comment by felixfontein** [comment]
@samccann yes, it should be there.

**Comment by samccann** [comment]
ok nvm found it - http://docs.testing.ansible.com/ansible/devel/collections/environment_variables.html (assuming some other test doesn't overwrite that).

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
@felixfontein final question and I think you told me before but I forgot  - does it take a PR after this is merged to get this new index someplace more visible? I had to hack the url to get to it cuz I couldn't find it otherwise...

**Comment by felixfontein** [comment]
@samccann right now this PR only makes sure that this new index can be found from http://docs.testing.ansible.com/ansible/devel/reference_appendices/config.html#environment-variables (and from any environment variable in plugin configs that links to it), there's currently no other link to it.

If someone has a good idea from where to link to it I can add it to this PR, otherwise it needs to be added in a follow-up PR.

**Comment by felixfontein** [comment]
I've added such a link in 5d4331cbb12cb8afa410b31b47feaf7113076ec7. The plural `s` for `collections` is not entirely appropriate for the ansible-core docs, since there is only one collection (ansible.builtin) (and yes, also for that one a bunch of env variables exists: https://docs.ansible.com/ansible-core/devel/collections/environment_variables.html).

**Comment by felixfontein** [comment]
@oraNod @samccann thanks a lot! I guess this should not be backported to stable-2.14?

**Comment by samccann** [comment]
@felixfontein  We try not to change requirements in the stable branches. Unless this is a serious bugfix, I think it stays in devel/2.15.

**Pull request merged**
